packages/flutter/lib/src/material/search.dart

```diff
@@ -276,8 +286,18 @@ abstract class SearchDelegate<T> {
   ///
   /// If this value is set to null, the value of the ambient [Theme]'s
   /// [InputDecorationTheme.hintStyle] will be used instead.
+  ///
+  /// If this value is not null, [searchFieldDecorationTheme]
```

Comment:
This statement doesn't match the constructor's assert. Best to just restate the fact that either this property or searchFieldDecorationTheme can be non-null.